File: src/test/java/com/raychenon/leetcode/SubarraySumEqualsKTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 27/4/2020
 * https://leetcode.com/problems/subarray-sum-equals-k
 */
public class SubarraySumEqualsKTest {

    private static SubarraySumEqualsK subarraySumEqualsK;

    @BeforeClass
    public static void setUp() {
        subarraySumEqualsK = new SubarraySumEqualsK();
    }

    @Test
    public void expect2Test() {
        evaluateMethods(2, new int[]{1, 1, 1}, 2);
    }

    @Test
    public void expect12Test() {
        evaluateMethods(12, new int[]{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}, 2);
    }

    @Test
    public void negativeNumbersTest() {
        evaluateMethods(2, new int[]{1, -2, 3, 1, 2, -3, 1}, -1);
    }

    void evaluateMethods(int expected, int[] nums, int k) {
        assertEquals(expected, subarraySumEqualsK.subarraySumBruteForce(nums, k));
        assertEquals(expected, subarraySumEqualsK.subarraySumDP(nums, k));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/SubarraySumEqualsKTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 27/4/2020
 * https://leetcode.com/problems/subarray-sum-equals-k/
 */
class SubarraySumEqualsKTest {

    @Test
    fun expect2Test() {
        evaluateMethods(2, intArrayOf(1, 1, 1), 2)
    }

    @Test
    fun expect12Test() {
        evaluateMethods(12, intArrayOf(1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1), 2)
    }

    @Test
    fun negativeNumbersTest() {
        evaluateMethods(2, intArrayOf(1, -2, 3, 1, 2, -3, 1), -1)
    }

    fun evaluateMethods(expected: Int, nums: IntArray, k: Int) {
        TestCase.assertEquals(
            expected, SubarraySumEqualsK.subarraySumBruteForce(nums, k)
        )
        TestCase.assertEquals(
            expected, SubarraySumEqualsK.subarraySumDP(nums, k)
        )
    }
}